Add push_to_hub docs

lhoestq opened the pull request:
Since #3098 it's now possible to upload a dataset on the Hub directly from python using the `push_to_hub` method.
I just added a section in the "Upload a dataset to the Hub" tutorial.

I kept the section quite simple but let me know if it sounds good to you @LysandreJik @stevhliu :)

mariosasko commented on docs/source/upload_dataset.rst:
```diff
@@ -77,6 +77,19 @@ Your dataset can now be loaded by anyone in a single line of code!
+Upload from Python
+------------------
+
+To upload a :class:`datasets.DatasetDict` on the Hugging face Hub in python, you can use the :func:`datasets.DatasetDict.push_to_hub` method:
```
One nit for consistency:

```suggestion
To upload a :class:`datasets.DatasetDict` on the Hugging Face Hub in Python, you can use the :func:`datasets.DatasetDict.push_to_hub` method:
```

stevhliu commented:
Looks good to me! :)

Maybe we can mention that users can also set the `private` argument if they want to keep their dataset private? It would lead nicely into the next section on Privacy.

lhoestq commented:
Thanks for your comments, I fixed the capitalization for consistency and added an passage to mention the `private` parameter and to have a nice transition to the Privacy section :)

I also added the login instruction that was missing before the user can actually upload a dataset.
